dapp/src/components/shared/TokenBalanceInput/index.tsx

```diff
@@ -118,27 +126,38 @@ export default function TokenBalanceInput({
   hasError = false,
   fiatCurrency,
   withMaxButton = false,
+  tokenAmountLabel = "Amount",
   ...inputProps
 }: TokenBalanceInputProps) {
   const valueRef = useRef<bigint | undefined>(amount)
   const styles = useMultiStyleConfig("TokenBalanceInput", { size })
 
-  const { decimals } = getCurrencyByType(currency)
+  const { decimals, symbol } = getCurrencyByType(currency)
 
   const handleValueChange = (value: string) => {
     valueRef.current = value ? userAmountToBigInt(value, decimals) : undefined
   }
 
+  const handleChange: ChangeEventHandler<HTMLInputElement> = (event) => {
+    inputProps.onChange?.(event)
+    setAmount(valueRef?.current)
```

Comment:
I was curious if `inputProps.onChange?.(event)` would affect the setting of the amount value. So I switched their order to tests.

After entering the value 5 , `inputProps.onChange?.(event)` changed the value to 0.0005. This is probably because our field is of non-standard bigint type and the event passes a string value to the `onChange` method. So I have some worries about calling `onChange` here. I think we should not allow the `onChange` function to be passed to `TokenBalanceInput`. I'm curious what you think.

https://github.com/user-attachments/assets/f3958cff-6597-41cc-ba8e-52ac69694610

Reply:
After changes from ea85581 the `onChange` handler call is redundant.
The explicit `onChange` call here was to ensure the change event is not lost in event propagation when I introduced the proxy event handler [here](https://github.com/thesis/acre/pull/687/commits/a72361a84776816593208a56d93e255c7640f83f#diff-69b290384051dd99cdfc367978535151273cc0eb90dcbf201bd20d03ee22b43dR32). Since I reset errors in `setAmount` callback  it's no longer valid.